kube-ovn-controller crashes: invalid memory address or nil pointer dereference

# Expected Behavior

# Actual Behavior

```txt
I0131 03:23:51.484560      13 leaderelection.go:248] attempting to acquire leader lease kube-system/kube-ovn-controller...
I0131 03:23:51.503124      13 leaderelection.go:258] successfully acquired lease kube-system/kube-ovn-controller
I0131 03:23:51.564858      13 controller.go:635] Starting OVN controller
I0131 03:23:51.565141      13 controller.go:642] Waiting for informer caches to sync
E0131 03:23:51.642806      13 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 493 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x55ae78633860?, 0x55ae79faf620})
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x55ae78a920e0?})
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/runtime/runtime.go:49 +0x75
panic({0x55ae78633860, 0x55ae79faf620})
	/usr/local/Cellar/go/1.19.5/libexec/src/runtime/panic.go:884 +0x212
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).podMatchNetworkPolicies(0xc000816800, 0xc0001c0000)
	/Users/zhangzujian/Alauda/kube-ovn/pkg/controller/network_policy.go:795 +0x11d
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).enqueueAddPod(0xc000816800, {0x55ae78a18a40?, 0xc0001c0000?})
	/Users/zhangzujian/Alauda/kube-ovn/pkg/controller/pod.go:66 +0xf4
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/Users/zhangzujian/go/pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/controller.go:232
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/Users/zhangzujian/go/pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/shared_informer.go:911 +0x134
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000618f38?, {0x55ae78a63760, 0xc000b49ef0}, 0x1, 0xc000ab4c00)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00005c048?, 0x3b9aca00, 0x0, 0x20?, 0xc000618f88?)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:92
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000138b00?)
	/Users/zhangzujian/go/pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/shared_informer.go:905 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:75 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:73 +0x85
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55ae774dbb5d]

goroutine 493 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x55ae78a920e0?})
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x55ae78633860, 0x55ae79faf620})
	/usr/local/Cellar/go/1.19.5/libexec/src/runtime/panic.go:884 +0x212
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).podMatchNetworkPolicies(0xc000816800, 0xc0001c0000)
	/Users/zhangzujian/Alauda/kube-ovn/pkg/controller/network_policy.go:795 +0x11d
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).enqueueAddPod(0xc000816800, {0x55ae78a18a40?, 0xc0001c0000?})
	/Users/zhangzujian/Alauda/kube-ovn/pkg/controller/pod.go:66 +0xf4
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/Users/zhangzujian/go/pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/controller.go:232
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/Users/zhangzujian/go/pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/shared_informer.go:911 +0x134
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000618f38?, {0x55ae78a63760, 0xc000b49ef0}, 0x1, 0xc000ab4c00)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00005c048?, 0x3b9aca00, 0x0, 0x20?, 0xc000618f88?)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:92
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000138b00?)
	/Users/zhangzujian/go/pkg/mod/k8s.io/client-go@v0.26.1/tools/cache/shared_informer.go:905 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:75 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/Users/zhangzujian/go/pkg/mod/k8s.io/apimachinery@v0.26.1/pkg/util/wait/wait.go:73 +0x85
```

# Steps to Reproduce the Problem

1. Create a ns and create a pod;
2. Create an empty ingress network policy in the ns;
3. Restart kube-ovn-controller.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  (paste your output here)
  ```

- kube-ovn version:
  
  ```
  master
  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**
  ```
  (paste your output here)
  ```

<!-- Any other additional information -->

--- a/pkg/controller/network_policy.go
+++ b/pkg/controller/network_policy.go
@@ -789,8 +789,20 @@ func isSvcMatchPod(svc *corev1.Service, pod *corev1.Pod) (bool, error) {
 }
 
 func (c *Controller) podMatchNetworkPolicies(pod *corev1.Pod) []string {
-	podNs, _ := c.namespacesLister.Get(pod.Namespace)
-	nps, _ := c.npsLister.NetworkPolicies(corev1.NamespaceAll).List(labels.Everything())
+	podNs, err := c.namespacesLister.Get(pod.Namespace)
+	if err != nil {
+		klog.Errorf("failed to get namespace %s: %v", pod.Namespace, err)
+		utilruntime.HandleError(err)
+		return nil
+	}
+
+	nps, err := c.npsLister.NetworkPolicies(corev1.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		klog.Errorf("failed to list network policies: %v", err)
+		utilruntime.HandleError(err)
+		return nil
+	}
+
 	match := []string{}
 	for _, np := range nps {
 		if isPodMatchNetworkPolicy(pod, *podNs, np, np.Namespace) {

